typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -24,111 +24,89 @@ export const checkValidatorSetup = async (
 
   let merkleTreeLatestCheckpointIndex: number | undefined;
   try {
-    const [_, latestCheckpointIndex] = await merkleTreeHook[
-      'latestCheckpoint()'
-    ]();
+    const [_, latestCheckpointIndex] = await merkleTreeHook.latestCheckpoint();
 
     merkleTreeLatestCheckpointIndex = latestCheckpointIndex;
     logBlue(
-      `\nLatest check point index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
+      `\nLatest checkpoint index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
     );
   } catch (err) {
     warnYellow(
       `❗️ Failed to fetch latest checkpoint index of merkleTreeHook on ${chain} \n`,
     );
   }
 
-  let announcedValidators;
+  const errorSet = new Set<string>();
+
+  const validatorsArray = Array.from(validators);
+  let validatorStorageLocations: string[][] | undefined;
+
   try {
-    announcedValidators = await validatorAnnounce.getAnnouncedValidators();
-  } catch (err) {
-    errorRed(
-      `❌ Failed to fetch announced validators for ${chain}. Exiting.\n`,
-    );
-    process.exit(1);
+    validatorStorageLocations =
+      await validatorAnnounce.getAnnouncedStorageLocations(validatorsArray);
+  } catch (e) {
+    errorSet.add('Failed to read announced storage locations on chain.');
```

Comment:
Surface a separate error for failing to read on-chain, put e in the message also @tkporter?